R/install_sys_deps.R

```diff
@@ -0,0 +1,186 @@
+
+  } else {
+    cat('Installing SRA Toolkit...\n')
+    # Change working directory
+    withr::local_dir(installDir)
+
+    # Determine latest available SRA Toolkit version
+    url = 'https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/'
+    raw = RCurl::getURL(url)
+    x = strsplit(raw, '\\n')[[1L]]
+    m = regexpr('[0-9]\\.[0-9]\\.[0-9]', x)
+    sraVersion = max(regmatches(x, m))
+
+    # Based on operating system, download correct version to correct path
+    sraOsTar = if (Sys.info()[['sysname']] == 'Darwin') 'mac64' else 'ubuntu64'
+    sraTar = glue('sratoolkit.{sraVersion}-{sraOsTar}.tar.gz')
+
+    # Download based on version and OS, unzip, then rename directory
+    withr::local_file(sraTar)
+    download.file(glue('{url}{sraVersion}/{sraTar}'), sraTar, quiet = TRUE)
+    utils::untar(sraTar)
+    . = file.rename(sub('\\.tar\\.gz', '', sraTar), 'sratoolkit')}
+
+  configPath = '~/.ncbi/user-settings.mkfg'
+  if (!file.exists(configPath)) {
+    if (!dir.exists(dirname(configPath))) dir.create(dirname(configPath))
+    uuidPath = withr::local_tempfile(fileext = '.txt')
+    download.file(
+      'https://www.uuidgenerator.net/api/version4', uuidPath, quiet = TRUE)
+    uuid = readLines(uuidPath, warn = FALSE)
+    prePath = system.file('extdata', 'user-settings.mkfg', package = 'seeker')
+    lines = sapply(
+      readLines(prePath, warn = FALSE), glue_data,
+      .x = list(uuid = uuid, home = Sys.getenv('HOME')))
+    writeLines(lines, configPath)}
+
+  if (addToPath) {
+    # Add to OS path and .Rprofile path
+    addToProfile(glue('export PATH="$PATH:{sraPath}"'))
+    path = paste(Sys.getenv('PATH'), sraPath, sep = ':')
+    addToProfile(glue('Sys.setenv(PATH = "{path}")'), type = 'R')
+    Sys.setenv(PATH = path)}
+
+  invisible()}
+
+
+installMiniconda = function(installDir, minicondaEnv, setSeekerOption = TRUE) {
+  # Determine paths for miniconda
+  minicondaPath = path.expand(file.path(installDir, 'miniconda3'))
+  minicondaEnvPath = if (minicondaEnv == 'base') {
+    minicondaPath
+  } else {
+    file.path(minicondaPath, 'envs', minicondaEnv)}
+
+  if (dir.exists(minicondaPath)) {
+    cat('Miniconda already installed, skipping...\n')
+
+  } else {
+    cat('Installing Miniconda...\n')
+
+    miniOsSh = if (Sys.info()[['sysname']] == 'Darwin') 'MacOSX' else 'Linux'
+    miniSh = glue('Miniconda3-latest-{miniOsSh}-x86_64.sh')
+    withr::local_file(miniSh)
+    download.file(
+      glue('https://repo.anaconda.com/miniconda/{miniSh}'), miniSh, quiet = TRUE)
+    system2('sh', c(miniSh, '-b', '-p', file.path(installDir, 'miniconda3')))
+
+    cat('Running conda init...\n')
+    system(glue('{minicondaPath}/bin/conda init bash'))}
+
+  # Create new environment if it doesn't exist
+  if (minicondaEnv != 'base' && !dir.exists(minicondaEnvPath)) {
+    cat('Creating conda environment...\n')
+    yamlPath = system.file('extdata', 'conda_env.yml', package = 'seeker')
+    envYaml = yaml::read_yaml(yamlPath)
```

Comment:
Do we want to rename these variables to be `condaYamlPath` and `condaEnvYaml` just for additional clarity, since we are using the mamba yamls as well?

Reply:
I think it's ok as is.